File: ws/collahuazo/u2/ws13GradesStudentAppCLI/JavaScript/model/grade.js
class Grade {
  constructor(studentId, value) {
    if (value < 0 || value > 20) {
      throw new Error("Grade must be between 0 and 20.");
    }
    this.studentId = studentId;
    this.value = value;
  }

  toJson() {
    return {
      studentId: this.studentId,
      value: this.value,
    };
  }

  static fromJson(data) {
    return new Grade(data.studentId || "", Number(data.value || 0));
  }
}

module.exports = Grade;

File: ws/collahuazo/u2/ws13GradesStudentAppCLI/Python/model/grade.py
class Grade:
    def __init__(self, student_id, value):
        if value < 0 or value > 20:
            raise ValueError("Grade must be between 0 and 20.")
        self.student_id = student_id
        self.value = value

    def to_dict(self):
        return {"studentId": self.student_id, "value": self.value}

    @staticmethod
    def from_dict(data):
        return Grade(data.get("studentId", ""), float(data.get("value", 0)))
